Rough draft of new Windows tutorial

davidwrede opened the pull request:
- Windows track is largely a copy of the Mac version with a few additions for using PowerShell.
- Includes conditional logic in the partial that adds Windows-specific content.
- New nav bar has been created and incorporated in site

Signed-off-by: David Wrede <dwrede@chef.io>

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

smurawski commented on www/source/shared/_create_plan_common.html.md.erb:
```diff
@@ -10,10 +12,17 @@ The Node.js application in this tutorial is a simple web app that displays a mes
        cd ~/habitat-example-plans/mytutorialapp
 
+<% if is_windows %>
+1. Change CRLF line endings to be LF line endings.
```
This will need to happen before the cloning of the mytutorial app.

davidwrede replied:
fixed

smurawski commented on www/source/tutorials/getting-started/windows/setup-environment.html.md.erb:
```diff
@@ -0,0 +1,41 @@
+
+2. Add the location for the `hab` CLI to your `PATH` environment variable.
+
+       $env:PATH = $env:PATH;$env:USERPROFILE/bin
```
this should be `$env:PATH = "$env:PATH;$env:USERPROFILE/bin"` (just missing the double quotes around the value to the right of the equals sign.

rrxtns replied:
no += to do add the necessary stuff? 

i.e. `$env:PATH += "$env:PATH;$env:USERPROFILE/bin"`

davidwrede replied:
This should now be fixed in the latest commit.

rrxtns commented on www/source/tutorials/getting-started/overview.html.md:
```diff
@@ -16,7 +16,7 @@ Before starting this tutorial, you need to have the following:
 *    An active GitHub account is recommended. If you don't already have an account, [sign up](https://github.com/) for one now. Note: This is required to upload and share your packages with others in the Habitat community.
 *   Your favorite text editor.
-*   If you are running Mac OS X on your host machine, then you need [Docker for Mac](https://www.docker.com/products/docker) installed. Make sure you have a Docker machine running before proceeding through the tutorial. 
+*   If you are running Mac OS X on your host machine, then you need [Docker for Mac](https://www.docker.com/products/docker#/mac) installed, and if you are running Microsoft Windows, you will need [Docker for Windows](https://www.docker.com/products/docker#/windows) installed. Make sure you have a Docker machine running before proceeding through the tutorial. 
```
I actually think this reads better if is formatted like so:

`*  If you are running Mac OS X`
`*  If you are running Microsoft Windows`

as separately line items, tis a minor nit though really.

davidwrede replied:
fixed

rrxtns commented:
Overall, a much better PR than mine - as the good captain says "[Engage!](http://gif.co/xjlI.gif)"

davidwrede commented:
@smurawski @rrxtns, I pushed new changes based on your feedback, so I'd love a quick review to make sure I've captured and addressed your concerns. I've also made a note in the overview page about recommending a shell emulator for people running older Win 10 builds. 

Let me know what you think.

rrxtns commented:
Only one reaction is necessary:

[Amazing!](http://gif.co/sBms.gif)

Full steam ahead!

davidwrede commented:
Thanks @rrxtns. :) 

Holding off on the merge until after the Windows binary goes out to Bintray.

mwrock commented on www/source/tutorials/getting-started/windows/create-plan.html.md.erb:
```diff
@@ -0,0 +1,61 @@
+
+       cd ~/habitat-example-plans/mytutorialapp/
+
+2. Make sure Docker for Mac is running and then enter into the studio environment.
```
Docker for Windows

smurawski replied:
got it in https://github.com/habitat-sh/habitat/pull/1350/commits/183b20d768742c1abd72192dc9cc26a1f2faa2ef

mwrock commented on www/source/tutorials/getting-started/windows/process-build.html.md.erb:
```diff
@@ -0,0 +1,72 @@
+
+    Habitat will proceed to unpack and install all necessary Habitat packages, including a Habitat supervisor, the mytutorialapp package, and all of its runtime dependencies. Then it will create an image using the Docker scratch image as the base and build up the rest of the image from there.
+
+4. Once that process has completed, exit out of the studio, make sure Docker for Mac is running and run your Docker image inside a container on your host machine.
```
Docker for Windows

smurawski replied:
got it in https://github.com/habitat-sh/habitat/pull/1350/commits/183b20d768742c1abd72192dc9cc26a1f2faa2ef

mwrock commented on www/source/tutorials/getting-started/windows/process-build.html.md.erb:
```diff
@@ -0,0 +1,72 @@
+
+    <%= partial "/shared/process_build_run_output" %>
+
+5. Because we are running this service in a Docker container on our host machine using Docker for Mac, you can simply connect to the Node.js app through your <a href="http://localhost:8080" target="_blank">localhost</a>.
```
for Windows

smurawski replied:
And this as well in https://github.com/habitat-sh/habitat/pull/1350/commits/183b20d768742c1abd72192dc9cc26a1f2faa2ef

smurawski commented on www/source/tutorials/getting-started/overview.html.md:
```diff
@@ -14,18 +14,19 @@ As a first step in understanding Habitat, this tutorial will show you how to cre
 *   Your favorite text editor.
-*   If you are running Mac OS X on your host machine, then you need [Docker for Mac](https://www.docker.com/products/docker) installed. Make sure you have a Docker machine running before proceeding through the tutorial. 
+*   If you are running Mac OS X on your host machine, then you need [Docker for Mac](https://www.docker.com/products/docker#/mac) installed.
+*   If you are running Microsoft Windows, you will need [Docker for Windows](https://www.docker.com/products/docker#/windows) installed. Also, if you are running an older Windows 10 version such as 1511, ANSI escape sequences are not supported. This means the color output and other formatting used by the `hab` CLI will not render properly in your PowerShell window. You can use console emulators l ...
```
Awesome!

mwrock commented:
New commits look, as the kids say, awesome like a possum!

smurawski commented:
@thesentinels approve

thesentinels commented:
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

thesentinels commented:
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/167662119).

thesentinels commented:
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/167662119).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

![gif-keyboard-3280869874741411265](https://cloud.githubusercontent.com/assets/4304/17755674/1e577b82-6490-11e6-96b7-1663d283c824.gif)
